src/lsp/cobol_ast/misc_descr.ml

```diff
@@ -14,25 +14,26 @@
 open Terms
 open Operands
 
-let if_empty pp_el pp_nel ppf = function
-  | [] -> pp_el ppf ()
-  | xs -> pp_nel ppf xs
-
-let elist ?prefix:(prf = Fmt.nop) ?sep pp =
-  if_empty Fmt.nop Fmt.(prf ++ list ?sep pp)
-
 
 (* -------------------- IDENTIFICATION DIVISION (EXTRA) -------------------- *)
 
 type informational_paragraphs =
-  {
-    author: string with_loc option;
-    installation: string with_loc option;
-    date_written: string with_loc option;
-    date_compiled: string with_loc option;
-    security: string with_loc option;
-  }
-[@@deriving show, ord]
+  informational_paragraph with_loc list
+[@@deriving ord]
+
+and informational_paragraph =
+  informational_paragraph_header * comment_entry with_loc
```

Comment:
What is the rational behind this change ? It's much easier to lookup the information in the record version than in the associated list, no ?

Maybe what we are struggling with is the fact that we chose to have only one AST to do everything, when we should probably have a "summarized" version of the AST: one AST that is close to the source, and one that only contains the interesting information for analyzis and compilation.